code/modules/mob/living/simple_animal/abnormality/waw/rose_sign.dm

```diff
@@ -0,0 +1,465 @@
+	var/list/work_roses = list()
+	var/list/work_damages = list()
+	var/rose_max = 4
+	var/rose_cooldown
+	var/rose_cooldown_time = 160 SECONDS
+	var/can_act = TRUE
+
+//*** Basic Simple mob procs***//
+/mob/living/simple_animal/hostile/abnormality/rose_sign/Move()
+	return FALSE
+
+/mob/living/simple_animal/hostile/abnormality/rose_sign/AttackingTarget(atom/attacked_target)
+	if(prob(30))
+		return OpenFire()
+	return
+
+/mob/living/simple_animal/hostile/abnormality/rose_sign/Life()
+	. = ..()
+	if((rose_cooldown < world.time) && !(status_flags & GODMODE))
+		PickTargets()
+		return
+
+/mob/living/simple_animal/hostile/abnormality/rose_sign/death()
+	density = FALSE
+	animate(src, alpha = 0, time = 10 SECONDS)
+	QDEL_IN(src, 10 SECONDS)
+	..()
+
+/mob/living/simple_animal/hostile/abnormality/rose_sign/WorkChance(mob/living/carbon/human/user, chance, work_type)//set the new work chances
+	var/rose_amount = LAZYLEN(work_roses)
+	if(rose_amount)
+		if(work_type == ABNORMALITY_WORK_INSIGHT)
+			return chance + (rose_amount * 4)
+		if(work_type == ABNORMALITY_WORK_INSTINCT && work_type == ABNORMALITY_WORK_ATTACHMENT)
+			return chance + (rose_amount * 8)
+		if(work_type == ABNORMALITY_WORK_REPRESSION)
+			return chance + (rose_amount * 15)
+	return chance
+
+/mob/living/simple_animal/hostile/abnormality/rose_sign/FailureEffect(mob/living/carbon/human/user, work_type, pe)
+	if(work_type == ABNORMALITY_WORK_INSIGHT || work_type == ABNORMALITY_WORK_REPRESSION)
+		return
+	datum_reference.qliphoth_change(-1)
+	return
+
+/mob/living/simple_animal/hostile/abnormality/rose_sign/PostWorkEffect(mob/living/carbon/human/user, work_type, pe)
+	if(work_type == ABNORMALITY_WORK_REPRESSION && !LAZYLEN(work_roses))
+		datum_reference.qliphoth_change(-1)
+		var/i = 1
+		var/list/speech_styles = list(SPAN_ITALICS, SPAN_ROBOT, SPAN_ITALICS, SPAN_SANS, SPAN_PAPYRUS, SPAN_ROBOT, SPAN_ITALICS)
+		var/list/lines = list("Foolish.", "Everyone has wishes. To deny them is to deny yourself.", "I can see how... vivid your seeds are. To not allow them to bloom isn't fair.", \
+		"Well... That's too bad.", "If you have no wishes, then remain here forevermore.", "Until you realize that there is no moving on without acceptance.", "Be with us. With me.")
+		for(var/line in lines)
+			speech_span = speech_styles[i]
+			say(line)
+			i++
+			sleep(15)
+		return
+	if(work_type == ABNORMALITY_WORK_INSIGHT && LAZYLEN(work_roses) >= rose_max)
+		datum_reference.qliphoth_change(-1)
+		var/i = 1
+		var/list/speech_styles = list(SPAN_ITALICS, SPAN_ROBOT, SPAN_SANS)
+		var/list/lines = list("Mm, still too pale.", "We don't need bland flowers like yours.", "How disappointing~")
+		for(var/line in lines)
+			speech_span = speech_styles[i]
+			say(line)
+			i++
+			sleep(15)
+		return
+	if(work_type == ABNORMALITY_WORK_INSIGHT)
+		SpawnWorkRose()
+	if(work_type == ABNORMALITY_WORK_REPRESSION)
+		var/rose = pick(work_roses)
+		work_roses -=  rose
+		qdel(rose)
+	return
```

Comment:
feel like this could be written better, specifically the speech.

Reply:
Loaded them all to the proc WorkSpeech which is on line 127 at the time of writing